velox/vector/tests/utils/VectorMaker.h

```diff
@@ -604,31 +630,120 @@ class VectorMaker {
   MapVectorPtr mapVector(
       const std::vector<std::vector<std::pair<TKey, std::optional<TValue>>>>&
           maps,
-      const TypePtr& type =
+      const TypePtr& mapType =
           MAP(CppToType<TKey>::create(), CppToType<TValue>::create())) {
+    std::vector<
+        std::optional<std::vector<std::pair<TKey, std::optional<TValue>>>>>
+        nullableMaps;
+    nullableMaps.reserve(maps.size());
+    for (const auto& m : maps) {
+      nullableMaps.push_back(m);
+    }
+
+    return mapVector<TKey, TValue>(nullableMaps, mapType, false);
+  }
+
+  template <typename TKey, typename TValue>
+  MapVectorPtr mapVector(
+      const std::vector<std::optional<
+          std::vector<std::pair<TKey, std::optional<TValue>>>>>& maps,
+      const TypePtr& mapType =
+          MAP(CppToType<TKey>::create(), CppToType<TValue>::create()),
+      bool hasNulls = true) {
     std::vector<vector_size_t> lengths;
     std::vector<TKey> keys;
     std::vector<TValue> values;
     std::vector<bool> nullValues;
+    std::vector<bool> nullRow;
+
     auto undefined = TValue();
 
     for (const auto& map : maps) {
-      lengths.push_back(map.size());
-      for (const auto& [key, value] : map) {
+      if (!map.has_value()) {
+        VELOX_CHECK(hasNulls);
+        nullRow.push_back(true);
+        lengths.push_back(0);
+        continue;
+      }
+      nullRow.push_back(false);
+      lengths.push_back(map->size());
+      for (const auto& [key, value] : map.value()) {
         keys.push_back(key);
         values.push_back(value.value_or(undefined));
         nullValues.push_back(!value.has_value());
       }
     }
 
+    std::function<bool(vector_size_t)> isNullAt = nullptr;
+    if (hasNulls) {
+      isNullAt = [&](vector_size_t row) { return nullRow[row]; };
+    }
+
     return mapVector<TKey, TValue>(
         maps.size(),
         [&](vector_size_t row) { return lengths[row]; },
         [&](vector_size_t idx) { return keys[idx]; },
         [&](vector_size_t idx) { return values[idx]; },
-        nullptr,
+        isNullAt,
         [&](vector_size_t idx) { return nullValues[idx]; },
-        type);
+        mapType);
+  }
+
+  /// Creates a MapVector from a list of JSON maps.
+  ///
+  /// JSON maps can represent a null map, an empty map or a map with null
+  /// values. Null keys are not allowed.
+  ///
+  /// Note that order of map keys in the MapVector is not guaranteed to match
+  /// the order of map keys in the JSON.
+  ///
+  /// Examples:
+  ///  {1: 10, 2: 20, 3: 30}
+  ///  {1: 10, 2: 20, 3: null, 4: 40}
+  ///  {1: null, 2: null}
+  ///  {} - empty map
+  ///  null - null map
+  ///
+  /// @tparam K Type of map keys. Must be an integer: int8_t, int16_t,
```

Comment:
shall we add static assert at least thats its not string or double since 
jsonValue wont throw at those,